pkg/runtime/wfengine/activity.go

```diff
@@ -78,19 +78,30 @@ func (a *activityActor) InvokeMethod(ctx context.Context, actorID string, method
 	}
 
 	// Try to load activity state. If we find any, that means the activity invocation is a duplicate.
-	if state, err := a.loadActivityState(ctx, actorID, ar.Generation); err != nil {
+	var state activityState
+	if _, err := a.loadActivityState(ctx, actorID, ar.Generation); err != nil {
 		return nil, err
 	} else if state.Generation > 0 {
 		return nil, ErrDuplicateInvocation
 	}
 
+	if methodName == "PurgeWorkflowState" {
+		if err := a.purgeActivityState(ctx, actorID, state); err != nil {
+			return nil, err
+		}
```

Comment:
Should this method return after processing this? We don't need to schedule a reminder, do we?